jax/_src/scipy/stats/multinomial.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import scipy.stats as osp_stats
+from jax import lax
+from jax._src.numpy import lax_numpy as jnp
+from jax._src.numpy.util import _wraps, _promote_args_inexact
+from jax._src.scipy.special import gammaln, xlogy
+
+@_wraps(osp_stats.multinomial.logpmf, update_doc=False)
+def logpmf(x, n, p):
+  """JAX implementation of scipy.stats.multinomial.logpmf."""
+  if not jnp.issubdtype(x.dtype, jnp.integer):
+    raise ValueError(f"x is of dtype {x.dtype}, it should be an integer.")
+  x = jnp.asarray(x, p.dtype)
+  n = jnp.asarray(n, p.dtype)
```

Comment:
Sorry to harp on this, but this still seems a bit off – for example, imagine someone puts in integer values for `p`. Then we'll convert everything to integer type, and the functions below will fail because they require floats.

There's also extra validation logic in `_promote_args_*` that is now missing from this implementation.

What if we do something like this instead:
```python
  p, = _promote_args_inexact("multinomial.logpmf", p)
  x, n = _promote_args_numeric("multinomial.logpmf", x, n)
  if not jnp.issubdtype(x.dtype, jnp.integer):
    raise ValueError(f"x and n must be of integer type; got dtype={x.dtype}")
  x = x.astype(p.dtype)
  n = n.astype(p.dtype)
```